core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1468,6 +1515,45 @@ public Builder setPreviousFileLocation(String previousFileLocation) {
       return this;
     }
 
+    private Builder setRowLineage(Boolean newRowLineage) {
```

Comment:
for builder, arg can probably be primitive `boolean`. then we can save the `if` check below

Reply:
This is used by the "enable row-lineage by table property" method, so it's possible that the value can be null